packages/@lwc/synthetic-shadow/src/3rdparty/polymer/path-composer.ts

```diff
@@ -19,24 +19,29 @@ import { isNull } from '@lwc/shared';
 import { getOwnerDocument } from '../../shared/utils';
 
 export function pathComposer(startNode: EventTarget, composed: boolean): EventTarget[] {
-    const composedPath: (Element | Document | Window)[] = [];
-    let current: Node | null = startNode as Element;
+    const composedPath: EventTarget[] = [];
+    let current: EventTarget | null = startNode;
     const startRoot: Window | Node =
         startNode instanceof Window ? startNode : (startNode as Node).getRootNode();
     while (!isNull(current)) {
         composedPath.push(current as Element);
```

Comment:
Do not need this type guard
```suggestion
        composedPath.push(current);
```